src/ui/header/Header.test.tsx

```diff
@@ -2,35 +2,47 @@ import { render } from 'tests';
 import { Header } from './Header';
 import { AppLocale } from 'context/locale/appLocale.enum';
 import { translations } from 'i18n/messages';
+import { AssertionError } from 'assert';
 
 describe('Header component', () => {
   it('renders language buttons correctly', () => {
-    const { getByText } = render(<Header />);
+    const { getByText } = render(<Header inputValue="" onSubmit={() => {}} />);
 
     expect(getByText(AppLocale.en)).toBeInTheDocument();
     expect(getByText(AppLocale.pl)).toBeInTheDocument();
   });
 
   it('has default active language button that is bold', () => {
-    const { getByText } = render(<Header />);
+    const { getByText } = render(<Header inputValue="" onSubmit={() => {}} />);
 
     expect(getByText(AppLocale.en)).toHaveClass('active');
     expect(getByText(AppLocale.en)).toHaveStyle('font-weight: 600');
   });
 
   it('has inactive language button that is not bold', () => {
-    const { getByText } = render(<Header />);
+    const { getByText } = render(<Header inputValue="" onSubmit={() => {}} />);
 
     expect(getByText(AppLocale.pl)).toHaveClass('not-active');
     expect(getByText(AppLocale.pl)).not.toHaveStyle('font-weight: 600');
   });
 
   it('changes language when language button, that is not actvie, is clicked', () => {
-    const { getByText, getAllByRole } = render(<Header />);
+    const { getByText, getAllByRole } = render(<Header inputValue="" onSubmit={() => {}} />);
 
     getByText(AppLocale.pl).click();
 
     expect(getByText(AppLocale.pl)).toHaveClass('active');
     expect(getAllByRole('button')[0]).toHaveTextContent(translations[AppLocale.pl]['header.searchButtonText']);
   });
+
+  it('displays input value', () => {
+    const { getByPlaceholderText } = render(<Header inputValue="sidecar 3,5t" onSubmit={() => {}} />);
+    const input = getByPlaceholderText(translations[AppLocale.en]['header.searchInputPlaceholder']);
+
+    if (!(input instanceof HTMLInputElement)) {
+      throw new AssertionError({ message: 'expected input to be an HTMLInputElement' });
```

Comment:
Mhm, nie mówię że to źle, ale co skłoniło, że wyrzucasz error w teście?

Reply:
Bez tego sprawdzenia czy input jest inputem, `input.value` zwracał mi błąd `Property 'value' does not exist on type 'HTMLElement'.`